components/text_sanitize/rs/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#[cxx::bridge(namespace = text_sanitize)]
+mod ffi {
+    extern "Rust" {
+        fn strip_non_alphanumeric_or_ascii_characters(text: &str) -> String;
+    }
+}
+
+// Replaces all non-alphanumeric, non-ascii graphic, or non-ascii whitespace
+// characters with whitespace.
+fn strip_non_alphanumeric_or_ascii_characters(text: &str) -> String {
+    let result: String = text
```

Comment:
You're not using the `result` local here, so it would be more idiomatic to yield the `collect()` chain directly without the `let`.
```rust
fn foo(..) -> String {
    text.chars()
        .map(|s| ...)
        .collect()
}
```